### PR TITLE
feat: initial gke-enterprise-cluster-observability dashboard

### PR DESCRIPTION
**Pull request by apeabody**
Adds a new `gke-enterprise-cluster-observability` folder and initial `gke-enterprise-cluster-observability-overview` dashboard

### CODE REVIEW

**Comment by apeabody** [comment]
FYI: @couragej

File: dashboards/google-kubernetes-engine-enterprise/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+sample_dashboards:
+  -
+    category: Google Kubernetes Engine
+    id: gke-enterprise-cluster-observability-overview
+    display_name: Google Kubernetes Engine (GKE) Enterprise Cluster Observability Overview
```
Comment (johnbryan) [inline dashboards/google-kubernetes-engine-enterprise/metadata.yaml:5]: Can this be "GKE Enterprise Cluster Observability Overview" to match the title in the json?

File: dashboards/google-kubernetes-engine-enterprise/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+sample_dashboards:
+  -
+    category: Google Kubernetes Engine
```
[the remark this answers is not in the document]
Comment (apeabody) [reply dashboards/google-kubernetes-engine-enterprise/metadata.yaml]: Sure @johnbryan - Changed to `Google Kubernetes Engine Enterprise`, should we also update the folder name?

File: dashboards/google-kubernetes-engine-enterprise/metadata.yaml (hunk 1 above)
Comment (apeabody) [reply dashboards/google-kubernetes-engine-enterprise/metadata.yaml]: Sure - Done!

**Review by johnbryan — APPROVED** [review]

**Pull request merged**